### ISSUE TITLE
.NET 6 NuGet packages for Android and iOS

### ISSUE THREAD

**Issue by akravch** [braze-inc/braze-xamarin-sdk#13]
Please provide NuGets that target `net6.0-android` and `net6.0-ios` - it's becoming really important now as Microsoft is abandoning classic Xamarin and moving to .NET 6.

**Comment by radixdev** [comment]
Hi @akravch ,

We've filed an issue to perform the upgrade. We will update this issue once we have more updates.

**Comment by akravch** [comment]
Do you guys have any ETA on this?

**Comment by radixdev** [comment]
We're still clearing out various Xamarin blockers to the .NET 6 upgrade but we definitely still have a keen eye on this upgrade and will share when that is about to be released. Apologies for the long delay on this!

**Comment by radixdev** [comment]
Hi @akravch ,

Today we released version 1.27.0 of the binding, which includes the android nuget binding for .NET-6 https://www.nuget.org/packages/BrazePlatform.BrazeAndroidBinding. Please let us know if you have any issues! We'll be releasing the iOS .NET-6 binding as well and will update this issue once that is live.

**Comment by xleon** [comment]
We need iOS package otherwise we can't migrate to MAUI unless we fork the project, which would be not very convenient

**Comment by radixdev** [comment]
Hello all,

We're actively developing our iOS .NET 6 binding and will update this issue once we have more updates. I appreciate your patience here!

**Comment by xleon** [comment]
Hi @radixdev any updates on this?

**Comment by radixdev** [comment]
Hi @xleon ,

No iOS updates at this time but we appreciate the patience on this and will update this issue once we have more updates.

**Comment by Vito258** [comment]
Hi, I'm a beginner with Braze. Can you please explain why the HandleCardUpdate method doesn't automatically execute when I use IContentCardsUpdateHandler, by creating an object and assigning it to the ContentCardUpdateHandler property of ContentCardsFragment?

**Comment by Vito258** [comment]
Hi @radixdev 
I'm a beginner with Braze. Can you please explain why the HandleCardUpdate method doesn't automatically execute when I use IContentCardsUpdateHandler, by creating an object and assigning it to the ContentCardUpdateHandler property of ContentCardsFragment?

**Comment by radixdev** [comment]
Hi @Vito258 ,

Please file a new issue with detailed steps to reproduce, sample code, and your desired usecase. From what you said, it sounds like the handler isn't being called since no Content Cards are being refreshed but let's dive in on a proper issue.

**Comment by zestfulyorghurt** [comment]
maui Braze Can IInAppMessageManagerListener be replaced with something

**Comment by zestfulyorghurt** [comment]
Hi @radixdev 
maui Braze Can IInAppMessageManagerListener be replaced with something

**Comment by jeremymarabel** [comment]
Hi @radixdev, I know you responded in september, but still no update on the ios binding ? 
This is becoming an issue, with xamarin end of support in the beginning of next year. Can you at least provide a approximate deadline for this ?

**Comment by hokstuff** [comment]
Hi all,

We are actively developing adding .NET 6+ support for iOS to NuGet (with the relevant XCFrameworks) and are nearing the final stages. Currently we don't have a specific timeline around this upcoming release, but we are aiming to have it completed in a matter of weeks / month(s). We will be sure to notify this thread with any updates.

Thank you all for your patience on this!

**Comment by gerardvanderkruijs** [comment]
Hi @hokstuff 
Thanks for the update!
Weeks/month(s) you say, if weeks I think this is acceptable, but month**s** hopefully not. Xamarin EOL is very close (May 2024) and everyone is upgrading their apps and packages or has already done so.
The only package holding us back now for finishing the migration is the Braze package.

So could you please give us a more accurate timeline?

Thanks in advance!

**Comment by hokstuff** [comment]
Hi @gerardvanderkruijs,

Just updating this thread - We have just completed the code and begun the internal review process for adding .NET 6+ and MAUI support. Due to the holiday, there may be a delay but it should be verified and released not long after. We are planning to release this as well as follow up with a release migrating to [the Swift SDK](https://github.com/braze-inc/braze-swift-sdk/). We are aware of the Xamarin EOL coming in May 2024 and will be sure to have this ready with additional lead-up time to migrate to the upcoming version.

Thanks!

**Comment by gerardvanderkruijs** [comment]
Hi @hokstuff 
Perfect! Thanks for the very fast reply and looking forward to it 🙌

**Comment by hokstuff** [comment]
Hi @akravch @xleon @Vito258 @zestfulyorghurt @jeremymarabel @gerardvanderkruijs,

We just released Xamarin SDK version 3.0.0 which includes support for .NET 6+ and .NET MAUI on iOS (Android support was added in a previous version). For iOS, you will need to follow the Breaking changes in the Changelog to import the new package. Feel free to contact support or comment here if you run into any issues.

Thanks!
